src/tests/test_utils/functional_test_utils/layer_tests_summary/run_parallel.py

```diff
@@ -282,156 +298,143 @@ def __get_test_list_by_runtime(self):
                 if pos > 0 or test_suite != "":
-                    real_test_name = test_suite + (test_name[2:pos-2] if pos > 0 else test_name[2:])
+                    real_test_name = test_suite + "." + (test_name[2:pos-1] if pos > 0 else test_name[2:])
                     if constants.DISABLED_PREFIX in real_test_name:
                         self._disabled_tests.append(real_test_name)
-                    else:
-                        test_list.append(f'"{self.__replace_restricted_symbols(real_test_name)}":')
+                    elif (test_unit == "test"):
+                        tests_dict[real_test_name] = 1
+                        self._total_test_cnt += 1
+                    elif (test_unit == "suite"):
+                        tests_dict[test_suite] = tests_dict.get(test_suite, 0) + 1
+                        self._total_test_cnt += 1
             test_list_file.close()
         os.remove(test_list_file_name)
-        logger.info(f"Len test_list_runtime (without disabled tests): {len(test_list)}")
-        if len(test_list) == 0:
+        logger.info(f"Len test_list_runtime (without disabled tests): {len(tests_dict)}")
+        if len(tests_dict) == 0:
             logger.warning(f"Look like there are not tests to run! Please check the filters!")
             exit(0)
-        return test_list
+        return tests_dict
 
     def __get_test_list_by_cache(self):
-        test_list_cache = list()
+        tests_dict_cache = dict()
         if os.path.isfile(self._cache_path):
             logger.info(f"Get test list from cache file: {self._cache_path}")
             with open(self._cache_path, "r") as cache_file:
                 for line in cache_file.readlines():
                     pos = line.find(":")
-                    time = line[:pos]
-                    test_name = line[pos+1:]
-                    if not constants.DISABLED_PREFIX in test_name:
-                        test_list_cache.append(TestStructure(test_name.replace("\n", ""), time))
-        logger.info(f"Len test_list_cache: {len(test_list_cache)}")
-        return test_list_cache
-
-    def __generate_test_lists(self, test_list_cache: list, test_list_runtime:list):
-        cached_test_list = list()
-        runtime_test_test = list()
-        cached_test_list_names = list()
-
-        for test in test_list_cache:
-            if test._name in test_list_runtime and not test in self._excluded_tests_re:
-                cached_test_list.append(test)
-                cached_test_list_names.append(test._name)
-        for test in test_list_runtime:
-            if not test in cached_test_list_names and not test in self._excluded_tests_re:
-                runtime_test_test.append(test)
-
-        if len(runtime_test_test) > 0:
+                    time = int(line[:pos])
+                    test_name = line[pos+1:].replace("\n", "")
+                    test_suite = test_name[:test_name.find(".")]
+
+                    if (self._split_unit == "test"):
```

Comment:
to define as constant -> "test" and "suite"

Reply:
fixed